Xamarin.Forms.Platform.iOS/Material/MaterialFrameRenderer.cs

```diff
@@ -1,165 +1,235 @@
 using System;
 using System.ComponentModel;
-using System.Drawing;
 using CoreAnimation;
+using CoreGraphics;
 using MaterialComponents;
 using UIKit;
 using Xamarin.Forms;
 using MCard = MaterialComponents.Card;
 
 [assembly: ExportRenderer(typeof(Xamarin.Forms.Frame), typeof(Xamarin.Forms.Platform.iOS.Material.MaterialFrameRenderer), new[] { typeof(VisualRendererMarker.Material) })]
+
 namespace Xamarin.Forms.Platform.iOS.Material
 {
-	public class MaterialFrameRenderer : MCard, IVisualElementRenderer
+	public class MaterialFrameRenderer : MCard,
+		IVisualElementRenderer
 	{
-		private VisualElementPackager _packager;
-		private VisualElementTracker _tracker;
+		nfloat _defaultElevation = -1f;
+		nfloat _defaultCornerRadius = -1f;
+		nfloat _defaultStrokeWidth = -1f;
+		UIColor _defaultBackgroundColor;
+		UIColor _defaultStrokeColor;
+
+		VisualElementPackager _packager;
+		VisualElementTracker _tracker;
+
+		public event EventHandler<VisualElementChangedEventArgs> ElementChanged;
 
 		public MaterialFrameRenderer()
 		{
 			VisualElement.VerifyVisualFlagEnabled();
 		}
 
-		Xamarin.Forms.Frame FrameElement
+		public Frame Element { get; private set; }
+
+		protected override void Dispose(bool disposing)
 		{
-			get { return Element as Xamarin.Forms.Frame; }
-		}
+			if (disposing)
+			{
+				if (_packager == null)
+					return;
 
-		public VisualElement Element { get; private set; }
+				SetElement(null);
 
-		public UIView NativeView
-		{
-			get { return this; }
+				_packager.Dispose();
+				_packager = null;
+
+				_tracker.NativeControlUpdated -= OnNativeControlUpdated;
+				_tracker.Dispose();
+				_tracker = null;
+			}
+
+			base.Dispose(disposing);
 		}
 
-		public UIViewController ViewController
+		public override void WillRemoveSubview(UIView uiview)
 		{
-			get { return null; }
+			var content = Element?.Content;
+			if (content != null && uiview == Platform.GetRenderer(content))
+			{
+				uiview.Layer.Mask = null;
```

Comment:
This may not be needed, but I want to make sure that we unhook everything that we hooked. This just empties the card's mask when the element is removed from the card.